File: discover_javascript/manipulate_the_data.js
var https = require('https');
var options = {
    hostname: 'api.github.com',
    path: '/search/repositories?q=language:javascript&sort=stars&order=desc',
    headers: {
	'User-Agent': 'Holberton_School',
	'Authorization': 'token bbb4a413d9e0340cd47db2f9b6898aadd5ea2cdc'
    }
}
var req = https.request(options, function(res) {
    streamToString(res,output);
});
req.end();

req.on('error', function(e) {
    console.error(e);
});

function streamToString(stream, cb) {
    const chunks = [];
    stream.on('data', (chunk) => {
	chunks.push(chunk);
    });
    stream.on('end', () => {
	cb(chunks.join(''));
    });
};

var output = function (jsonString)
{
      /* console.log(typeof jsonString);                                                                                                                             
	  console.log(jsonString);*/
    const fs = require('fs');
    fs.writeFile("/tmp/36", jsonString, function(err) {
	if (err) {
	    return console.log(err);
	}
    });
    var obj = jsonString.items[full_name];
    console.log(JSON.parse(obj));    
}
                                              

File: discover_more_python/manipulate_the_data.py
import urllib2
import urllib
import json

request_headers = {
  'User-Agent': 'Holberton_School',
  'Authorization': 'token 6be002d3b7968b5ef3fbbb318e97afe71225afe5'
}  

url = 'https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc'
req = urllib2.Request(url, None, request_headers)
response = urllib2.urlopen(req)
the_page = response.read()
parsed_json = json.loads(the_page)

for i in parsed_json['items']:
    print i['full_name']